src/highrl/utils/teacher_checker.py

```diff
@@ -1,5 +1,8 @@
 """Difficulty computation implementation for the teacher"""
 from typing import Tuple, List
+import logging
+import time
+from collections import deque
```

Comment:
Why use deque?

Reply:
I am using the `deque` here as a `queue`. The main benefit is the speed, the queue has two main methods:
- `enqueue` 
- `dequeue`, this method is `O(1)` instead of the old version `O(n)`.

Reply:
Noted